access/types/stdasync.py

```diff
@@ -330,20 +321,16 @@ def acceptGeneralForm(request, course, exercise, post_url):
         exercise
     )    
 
-
-def _requireActions(exercise):
-    '''
-    Checks that some actions are set.
-    '''
-    if "actions" not in exercise or len(exercise["actions"]) == 0:
-        if not settings.CONTAINER_MODE or "container" not in exercise:
-            raise ConfigError("Missing \"actions\" in exercise configuration.")
-
-
 def _requireContainer(exercise):
     c = exercise.get("container", {})
+    a = exercise.get("actions", {})
+    
+    if a:
```

Comment:
This should be checked only if there is no container configuration. I.e. if there is no container, check if there is actions and edit the warning to state that. If there is neither, then so the default error.

It's very common that actions are left in the configs and this change would break many courses for now. Of course, courses should remove actions. Best is to start raising warnings in build process, which is not easy currently, and kind of is waiting for roman/shepherd.

Reply:
Yes, It is true. I will change the `raise ConfigError` for a `LOGGER.warning`